submissions/sofiichuk/frogger-task/app.js

```diff
@@ -0,0 +1,147 @@
+const playingField = {
+    width: 500,
+    height: 600
+};
+
+const cell = {
+    width: 100,
+    height: 85
+};
+
+const river = {
+    height: 85
+};
+
+const character = {
+    width: 60,
+    height: 60
+};
+
+const initialPlayerPosition = {
+    x: 200,
+    y: 385
+};
+
+let count = 0;
+
+// ===========================================================
+
+const score = document.createElement('h3');
+score.textContent = 'your score is: 0';
+document.body.append(score);
+
+// ===========================================================
+
+const Enemy = function (x, y, speed) {
+    this.x = x;
+    this.y = y;
+    this.speed = speed;
+    this.sprite = 'images/enemy-bug.png'
+};
+
+Enemy.prototype.update = function (dt) {
+    if (this.x <= playingField.width) {
+        this.x = this.speed * dt + this.x;
+    } else {
+        this.x = 0;
+    }
+    this.detectCollision();
+};
+
+Enemy.prototype.detectCollision = function () {
+    if (player.x <= this.x + character.width &&
+        player.x + character.width >= this.x &&
+        player.y <= this.y + character.height &&
+        player.y + character.height >= this.y) {
+        player.reset()
```

Comment:
Item 4.iii of the requirements.

Reply:
replaced global variable's properties with enemy's own